### ISSUE TITLE
Upgrade ACS from v4.0 to v4-latest

### ISSUE THREAD

**Issue by wmhutchison** [BCDevOps/developer-experience#4389]
**Describe the issue**
A significant amount of work has been sunk into upgrading ACS from v3.74 to v4.0, mainly due to the complexity and nature of the upgrade involved (a significant DB upgrade/change). However v4.0 is not the latest version of ACS by far, and thus we need to spec out the requirements (time, complexity) involved with getting ACS upgraded fully to its latest supported version. This ticket hopes to track all of that.

**Additional context**
Increased priority due to learning that ACS v4.0 is already past EOL, so we need to push forward on upgrading LAB/PROD ACS as quickly as possible.

**How does this benefit the users of our platform?**
Newer ACS versions mean newer functionality and bugfixes which might not be available in older ACS releases.

**Definition of done**
- [x] test and document the work involved to fully upgrade the LAB ACS environment in KLAB.
- [x] Schedule and execute suitable RFC for doing the same for PROD ACS.

**Comment by wmhutchison** [comment]
LAB ACS upgraded now complete from v4.0 to v4.2. LAB CCM app rhacs paused for the time being, will only be unpaused when all ACS upgrades are completed in LAB/PROD and a new CCM PR pushed to formalize the final channel setting.

**Comment by wmhutchison** [comment]
Copy/pasting below the proposed PROD ACS methodology. Will create said RFC and comms tomorrow, aiming for execution starting November 21st.

```
1. Check the status of all member clusters in ACS web console. Do not attempt upgrade unless all clusters involved show a Cluster Status of Healthy for all components. https://acs.developer.gov.bc.ca/main/clusters

2. Take a backup of key components of the ACS cluster on SILVER in case it is needed for roll-back.
  - backup Cluster init bundle.
     - login to ACS web console (https://acs.developer.gov.bc.ca/) with either admin u/p or as a user with cluster-admin privileges in Openshift.
     - navigate to `Platform Configuration->Clusters->Manage Tokens`.
     - select `Cluster Init Bundle`, then click `Generate bundle`.
     - a name will need to be given to the contents, name it accordingly like `2023-prod-acs-v42`.
     - a download link will appear for `Kubernetes secrets file`. Click it to download the resulting YAML to your computer.
     - use standard methods to copy the downloaded file over to the SILVER-UTIL server and place the file in the /tmp/stackrox_bu_v42 directory.
  - backup ACS database. Remove token.txt once done.
      ```
      cd /tmp/stackrox__bu_v42
      vi token.txt (put contents of API token in here)
      export ROX_API_TOKEN=$(cat token.txt)
      export ROX_CENTRAL_ADDRESS=acs.developer.gov.bc.ca:443
      roxctl -e "$ROX_CENTRAL_ADDRESS" central backup
      ```
  - backup deployment using RHACS Operator.
      ```
      cd /tmp/stackrox_backup
      oc -n stackrox get central central -o yaml > central-cr.yaml
      oc get secret -n stackrox central-tls -o json | jq 'del(.metadata.ownerReferences)' > central-tls.json
      oc get secret -n stackrox central-htpasswd -o json | jq 'del(.metadata.ownerReferences)' > central-htpasswd.json
      ```
3. Pause sync on all PROD rhacs CCM apps. This will not be removed during this RFC, instead will be dealt with in a future CCM release.

4. Upgrade the entire ACS cluster twice, first upgrading all ACS clusters from v4.0 to v4.1, and then from v4.1 to v4.2 via the following list of instructions.

   4a. Set ACS operator channel on SILVER to the target version we want to upgrade to (rhacs-4.1 or rhacs-4.2 depending on current version).

   4b. Approve new installplan when it is ready.

   4c. Monitor pods in SILVER stackrox namespace to ensure nothing is crashlooping. Monitor cluster status as well via https://acs.developer.gov.bc.ca/main/clusters

   4d. when cluster status for SILVER via mentioned URL shows all green except for Collector, then proceed to step 4e.

   4e. For all member clusters (GOLD/GOLDDR/EMERALD), change their ACS subscription channel to match that for SILVER and then monitor progress via stackrox namespace on the clusters and SILVER's web console URL for showing cluster status as seen from Central.
   4f. When all clusters in the console URL (https://acs-lab.developer.gov.bc.ca/main/clusters) show all green, then the version upgrade is complete. Double-check stackrox pods in all involved clusters as well to ensure pod churn has resided.
   4g. If there is another upgrade to perform (have upgraded to v4.1 now need to upgrade to v4.2) then repeat all of the components in step 4 with this new version in mind.

5. At the end, change the channel for ACS on SILVER/GOLD/GOLDDR/EMERALD from to "stable" which going forward will keep us organically upgraded to the newest version without further invervention.

6. If step 5 generates new installplans due to a minor new version being released, then approve these by doing so first for SILVER and making sure the web cluster URL is green before doing the rest.

7. Notify stakeholders via email and Rocket Chat regarding completion.
```

**Comment by wmhutchison** [comment]
CHG0054166 was successfully scheduled and implemented to upgrade ACS to v4.2. Future ACS upgrades will be handled as part of regular CCM updates. If a LAB ACS upgrade shows evidence of possible significant disruption to PROD then PROD ACS upgrade will be covered by RFC again.
